src/agent/rustjail/src/cgroups/fs/mod.rs

```diff
@@ -174,7 +174,9 @@ impl CgroupManager for Manager {
                 freezer_controller.freeze()?;
             }
             _ => {
-                return Err(anyhow!(nix::Error::EINVAL));
+                return Err(anyhow!(
+                    "Invalid FreezerState. Must be FreezerState::Frozen or FreezerState::Thawed"
```

Comment:
I'd consider inverting this as although the message is superficially helpful, it doesn't result in the actual state being logged, so doesn't help developers. I'm being arguably overly harsh/pedantic here as there is only one other state, but what if there are >1 other states in future?

I'd just log something like:

```suggestion
                    "Invalid FreezerState: {:?}", state
```

Of course, that won't currently work as FreezerState` doesn't support `Display`/`Debug`, but I think it should. If you're keen, you might want to raise a PR on https://github.com/kata-containers/cgroups-rs to add atleast `#[derive(Debug)]` to the `enum`.